src/screens/Ticketing/Ticket/TicketInfo.tsx

```diff
@@ -92,17 +112,27 @@ const groupByUserProfileNames = (
       return [...groupedById, {userProfileId: t.user_profile_ref, count: 1}];
     }, [] as {userProfileId: string; count: number}[])
     .map((idAndCount) => {
-      const userProfile = findById(userProfiles, idAndCount.userProfileId);
-      const name = userProfile
-        ? getReferenceDataName(userProfile, language)
-        : undefined;
+      const userProfile = findReferenceDataById(
+        userProfiles,
+        idAndCount.userProfileId,
+      );
       return {
-        name,
+        ...userProfile,
         count: idAndCount.count,
       };
     })
     .filter(
-      (nameAndCount): nameAndCount is NameAndCount => nameAndCount.name != null,
+      (userProfileWithCount): userProfileWithCount is UserProfileWithCount =>
```

Comment:
is this mainly to get typing correct or am I not reading this correct? are there cases where mapped data from above does not contain ID?